packages/core-data/src/resolvers.js

```diff
@@ -321,24 +321,28 @@ export const getEntityRecords =
 						} ) );
 
 					const canUserResolutionsArgs = [];
+					const receiveUserPermissionArgs = [];
 					for ( const targetHint of targetHints ) {
 						for ( const action of ALLOWED_RESOURCE_ACTIONS ) {
 							canUserResolutionsArgs.push( [
 								action,
 								{ kind, name, id: targetHint.id },
 							] );
 
-							dispatch.receiveUserPermission(
+							receiveUserPermissionArgs.push( [
 								getUserPermissionCacheKey( action, {
 									kind,
 									name,
 									id: targetHint.id,
 								} ),
-								targetHint.permissions[ action ]
-							);
+								targetHint.permissions[ action ],
+							] );
 						}
 					}
 
+					dispatch.receiveUserPermissions(
+						receiveUserPermissionArgs
```

Comment:
There should be `Object.fromEntries( receiveUserPermissionArgs )`, otherwise the reducer will store garbage into the `userPermissions` state.

Reply:
@jsnajdr Can you tell me what the difference is?

Reply:
Resolved in [062a31d](https://github.com/WordPress/gutenberg/pull/64894/commits/062a31dc1964025a208904ff557a47e03d0ec8d2).